openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -647,116 +602,109 @@ def _get_library_component_tags_count(library_key) -> dict:
+        raise ContentLibraryBlockNotFound(usage_key) from exc
+
+    # The component might have existed at one point, but no longer does because
+    # the draft was soft-deleted. This is actually a weird edge case and I'm not
+    # clear on what the proper behavior should be, since (a) the published
+    # version still exists; and (b) we might want to make some queries on the
+    # block even after it's been removed, since there might be versioned
+    # references to it.
+    draft_version = component.versioning.draft
+    if not draft_version:
+        raise ContentLibraryBlockNotFound(usage_key)
+
+    published_version = component.versioning.published
+
     return LibraryXBlockMetadata(
         usage_key=usage_key,
-        def_key=def_key,
-        display_name=get_block_display_name(def_key),
-        has_unpublished_changes=lib_bundle.does_definition_have_unpublished_changes(def_key),
+        display_name=draft_version.title,
+        has_unpublished_changes=(draft_version != published_version),
     )
 
 
-def get_library_block_olx(usage_key):
-    """
-    Get the OLX source of the given XBlock.
-    """
-    assert isinstance(usage_key, LibraryUsageLocatorV2)
-    definition_key = get_library_block(usage_key).def_key
-    xml_str = get_bundle_file_data(
-        bundle_uuid=definition_key.bundle_uuid,  # pylint: disable=no-member
-        path=definition_key.olx_path,  # pylint: disable=no-member
-        use_draft=DRAFT_NAME,
-    ).decode('utf-8')
-    return xml_str
-
-
 def set_library_block_olx(usage_key, new_olx_str):
     """
     Replace the OLX source of the given XBlock.
+
     This is only meant for use by developers or API client applications, as
     very little validation is done and this can easily result in a broken XBlock
     that won't load.
     """
     # because this old pylint can't understand attr.ib() objects, pylint: disable=no-member
     assert isinstance(usage_key, LibraryUsageLocatorV2)
+
     # Make sure the block exists:
-    metadata = get_library_block(usage_key)
-    block_type = usage_key.block_type
-    # Verify that the OLX parses, at least as generic XML:
+    _block_metadata = get_library_block(usage_key)
+
+    # Verify that the OLX parses, at least as generic XML, and the root tag is correctd:
     node = etree.fromstring(new_olx_str)
-    if node.tag != block_type:
-        raise ValueError(f"Invalid root tag in OLX, expected {block_type}")
-    # Write the new XML/OLX file into the library bundle's draft
-    draft = get_or_create_bundle_draft(metadata.def_key.bundle_uuid, DRAFT_NAME)
-    write_draft_file(draft.uuid, metadata.def_key.olx_path, new_olx_str.encode('utf-8'))
-    # Clear the bundle cache so everyone sees the new block immediately:
-    BundleCache(metadata.def_key.bundle_uuid, draft_name=DRAFT_NAME).clear()
+    if node.tag != usage_key.block_type:
+        raise ValueError(
+            f"Tried to set the OLX of a {usage_key.block_type} block to a <node.tag> node. "
```

Comment:
I don't think we want to print a literal `<node.tag>` here? You had `<{node.tag}>` before 12f138b5fe810cd622090ba197cdc3bc499292d6, which makes more sense to me.

Reply:
Whoops, yup, t